web/src/pages/insights/DetailsTable.tsx

```diff
@@ -0,0 +1,202 @@
+    }
+}
+
+const getCellValue = (details: ProjectInsightsDetails, key: ColumnKey): React.ReactNode => {
+    if (key === 'stripy' || key === 'mito') {
+        const report = details.web_reports?.[key]
+        return report ? <a href={(report as { url: string }).url}>Link</a> : 'N/A'
+    }
+
+    const value = details[key as keyof ProjectInsightsDetails]
+
+    if (typeof value === 'boolean') {
+        return value ? '✅' : '❌'
+    }
+
+    return value
+}
+
+const DetailsTableRow: React.FC<{ details: ProjectInsightsDetails }> = ({ details }) => {
+    const rowClassName = getRowClassName(details.sequencing_type)
+    return (
+        <SUITable.Row key={`${details.sequencing_group_id}`} className={rowClassName}>
+            {detailsTableHeaderCellConfigs.map((config) => (
+                <SUITable.Cell
+                    key={config.key}
+                    data-cell
+                    className={
+                        config.key === 'dataset' || config.key === 'sequencing_type'
+                            ? 'category-cell'
+                            : 'table-cell'
+                    }
+                >
+                    {getCellValue(details, config.key)}
+                </SUITable.Cell>
+            ))}
+        </SUITable.Row>
+    )
+}
+
+const DetailsTable: React.FC<DetailsTableProps> = ({
+    filteredData,
+    handleSelectionChange,
+    getUniqueOptionsForColumn,
+    getSelectedOptionsForColumn,
+}) => {
+    const [sortColumns, setSortColumns] = useState<
+        Array<{ column: ColumnKey; direction: 'ascending' | 'descending' }>
+    >([])
+
+    const handleSort = (column: ColumnKey, isMultiSort: boolean) => {
+        if (isMultiSort) {
+            const existingColumnIndex = sortColumns.findIndex(
+                (sortColumn) => sortColumn.column === column
+            )
+
+            if (existingColumnIndex !== -1) {
+                const updatedSortColumns = [...sortColumns]
+                updatedSortColumns[existingColumnIndex].direction =
+                    updatedSortColumns[existingColumnIndex].direction === 'ascending'
+                        ? 'descending'
+                        : 'ascending'
+                setSortColumns(updatedSortColumns)
+            } else {
+                setSortColumns([...sortColumns, { column, direction: 'ascending' }])
+            }
+        } else {
+            if (sortColumns.length === 1 && sortColumns[0].column === column) {
+                setSortColumns([
+                    {
+                        column,
+                        direction:
+                            sortColumns[0].direction === 'ascending' ? 'descending' : 'ascending',
+                    },
+                ])
+            } else {
+                setSortColumns([{ column, direction: 'ascending' }])
+            }
+        }
+    }
+
+    const sortedData = React.useMemo(() => {
+        const data = [...filteredData]
+        data.sort((a, b) => {
```

Comment:
Maybe consider using the lodash orderBy method, it allows multiple column sort, and returns the new sorted array for you. Though I think it uses the javascript sort operator

Reply:
lodash orderBy looks really well suited for this. We can pass the column names as an array, sort directions `'asc'`/`'desc'` array, and get the sorted array out. Seems really simple.

However it got a bit tricky in practice because the sorting logic is tied up in the semantic-ui-react `Table.HeaderCell` component, which has a `sorted` property that only allows the values `'ascending'` and `'descending'`. 

As far as I can tell, the sorting is mostly functional as it stands. I agree that using lodash orderBy will be more optimal, but I might leave that for the next PR if it's not too big a deal. Cheers